client/my-sites/post-share/index.jsx

```diff
@@ -123,7 +140,26 @@ class PostShare extends Component {
 	};
 
 	sharePost = () => {
-		this.props.sharePost( this.props.siteId, this.props.postId, this.state.skipped, this.state.message );
+		const {
+			postId,
+			siteId,
+			connections,
+		} = this.props;
+		if ( this.state.scheduledDate ) {
+			const servicesToPublish = difference(
+				connections.map( connection => connection.keyring_connection_ID ),
+				this.state.skipped
+			);
+			this.props.schedulePostShareAction(
+				siteId,
+				postId,
+				this.state.message,
+				this.state.scheduledDate.format( 'X' ),
+				servicesToPublish,
+			);
+		} else {
+			this.props.sharePost( this.props.siteId, this.props.postId, this.state.skipped, this.state.message );
```

Comment:
Since some of these props have been deconstructed into `postId`, `site_id` above, why not deconstruct all of them and just pass them raw here? Or at least we can use those two.